src/components/Chains/ChainSelector/index.tsx

```diff
@@ -0,0 +1,100 @@
+
+import clsx from 'clsx';
+
+import Select, {
+  SelectButton,
+  SelectOptions,
+  SelectOption,
+  SelectBody,
+  SelectCheck,
+  SelectText,
+  SELECT_VARIANTS
+} from 'components/Select';
+
+interface ChainOption {
+  name: string;
+  icon: JSX.Element;
+}
+
+interface Props {
+  chainOptions: Array<ChainOption>;
+  currentChain: ChainOption;
+  onChange: (chain: ChainOption) => void;
+}
+
+const ChainSelector = ({
+  chainOptions,
+  currentChain,
+  onChange
+}: Props): JSX.Element => (
+  <Select
+    variant={SELECT_VARIANTS.formField}
+    key={currentChain.name}
+    value={currentChain}
+    onChange={onChange}>
+    {({ open }) => (
+      <>
+        <SelectBody
+          className={clsx(
+            'w-full'
+          )}>
+          <SelectButton variant={SELECT_VARIANTS.formField}>
+            <span
+              className={clsx(
+                'flex',
+                'items-center',
+                'space-x-3',
+                'text-xl',
+                'py-2'
+              )}>
+              {currentChain.icon}
+              <SelectText>
+                <span className='capitalize'>
```

Comment:
@tomjeatt 
I don't think we need this additional `span`. It's redundant because `SelectText` has className prop exposed.